src/xcode/ENA/ENA/Source/Services/TicketValidation/AllowList/AllowListService.swift

```diff
@@ -0,0 +1,127 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import ENASecurity
+
+final class AllowListService {
+	
+	// MARK: - Init
+	
+	init(
+	    restServiceProvider: RestServiceProviding,
+	    store: Store
+	) {
+		self.restServiceProvider = restServiceProvider
+		self.store = store
+	}
+	
+	// MARK: - Private
+
+	private let restServiceProvider: RestServiceProviding
+	private let store: Store
+	
+	// MARK: - Internal
+
+	func fetchAllowList() {
+		let resource = AllowListResource()
+		restServiceProvider.load(resource) { [weak self] result in
+			guard let self = self else { return }
+
+			switch result {
+			case .success(let allowList):
+				let serviceProviderAllowlist = allowList.serviceProviders.map({
+					$0.serviceIdentityHash
+				})
+				let validationServiceAllowlist = allowList.certificates.map({
+					ValidationServiceAllowlistEntry(
+						serviceProvider: $0.serviceProvider,
+						hostname: $0.hostname,
+						fingerprint256: $0.fingerprint256.base64EncodedString()
+					)
+				})
+				self.store.ticketValidationAllowList = TicketValidationAllowList(
+					validationServiceAllowList: validationServiceAllowlist,
+					serviceProviderAllowList: serviceProviderAllowlist
+				)
+				
+			case .failure(let error):
+				Log.debug(error.localizedDescription, log: .ticketValidationAllowList)
+			}
+		}
+	}
+	
+	func checkServerCertificateAgainstAllowlist(
+		hostname: String,
+		certificateChain: [Data], // can be construct by: Data(base64Encoded: x509String)
+		allowlist: [ValidationServiceAllowlistEntry]
+	) -> Result<Void, AllowListError> {
+		guard let leafCertificate = certificateChain.first else {
+			Log.debug("Certificate chain should include at least one certificate", log: .ticketValidationAllowList)
+			return .failure(.CERT_PIN_MISMATCH)
+		}
+		// Find requiredFingerprints: the requiredFingerprints shall be set by mapping each entry in allowlist to their fingerprint256 attribute.
+
+		let requiredFingerprints = allowlist.map({
+			$0.fingerprint256
+		})
+		
+		// Compare fingerprints: if the SHA-256 fingerprints of leafCertificate is not included in requiredFingerprints, the operation shall abort with error code CERT_PIN_MISMATCH.
+		let leafFingerPrint = leafCertificate.sha256().base64EncodedString()
```

Comment:
Fingerprint is one word

```suggestion
		let leafFingerprint = leafCertificate.sha256().base64EncodedString()
```

Reply:
The naming is correct as we are following the techspec convention here, please check TrustEvaluation.swift as well

Reply:
Then we should follow the same convention in line 72. 

It should be then requiredFingerPrints